pkg/controller/restore_controller.go

```diff
@@ -247,6 +251,14 @@ func (controller *restoreController) processRestore(key string) error {
 		}
 	}
 
+	// TODO: What do we do with the error here? It'll get handled as a validation error later,
+	// but if we need a schedule name for the restore before that, we're throwing our metrics off.
+	backup, _ := controller.fetchBackup(controller.bucket, restore.Spec.BackupName)
```

Comment:
The error is ignored for now as I want to get some input on this.

We're getting the backup itself in 2 functions below here - `getValidationErrors` and `runRestore`. Neither of those returns the backup fetched up to this `processRestore` function.

By calling `fetchBackup` here, we're duplicating one of the validation checks (if the backup even exists). Should we do that here? If the backup doesn't exist, we can't get a schedule name, which could skew the number of attempts towards the `schedule=""` label;  I'm not sure we can do much else in this case.

Reply:
This code is in need of a cleanup, but that'll be a broader effort to move as close to 100% of the defaulting, validation, and processing logic from the controller to the restorer (and the same for backups). In the meantime:

I would recommend that we call `controller.fetchBackup` one time: in `processRestore`, immediately before we call `controller.getValidationErrors`. Let's store the fetched backup in a variable, and pass it - `getValidationErrors(restore, backup)`. Let's also pass it to `runRestore`.

As for the metrics, and potentially skewing the `schedule=""` label if we can't find the backup, that's probably ok, at least for now, and hopefully for the long term too. We _could_ try to parse the name of the backup to see if it looks like it has a timestamp suffix, but let's not worry about that for now.